src/codegen/spirv.zig

```diff
@@ -2223,18 +2232,52 @@ const DeclGen = struct {
         comptime modular: bool,
     ) !?IdRef {
         if (self.liveness.isUnused(inst)) return null;
+
         // LHS and RHS are guaranteed to have the same type, and AIR guarantees
         // the result to be the same as the LHS and RHS, which matches SPIR-V.
         const ty = self.typeOfIndex(inst);
         const bin_op = self.air.instructions.items(.data)[inst].bin_op;
-        var lhs_id = try self.resolve(bin_op.lhs);
-        var rhs_id = try self.resolve(bin_op.rhs);
-
-        const result_ty_ref = try self.resolveType(ty, .direct);
+        const lhs_id = try self.resolve(bin_op.lhs);
+        const rhs_id = try self.resolve(bin_op.rhs);
 
         assert(self.typeOf(bin_op.lhs).eql(ty, self.module));
         assert(self.typeOf(bin_op.rhs).eql(ty, self.module));
 
+        return self.arithOp(ty, lhs_id, rhs_id, fop, sop, uop, modular);
+    }
+
+    fn arithOp(
+        self: *DeclGen,
+        ty: Type,
+        lhs_id_: IdRef,
+        rhs_id_: IdRef,
+        comptime fop: Opcode,
+        comptime sop: Opcode,
+        comptime uop: Opcode,
+        /// true if this operation holds under modular arithmetic.
+        comptime modular: bool,
+    ) !?IdRef {
```

Comment:
This function no longer needs to return an optional anymore, it can be just `!IdRef`